graphql/resolvers/alumnis.js

```diff
@@ -118,5 +121,116 @@ module.exports = {
 
       return newAlumni;
     },
+
+    async editAlumniProfile(
+      _,
+      {
+        editAlumniProfileInput: {
+          firstName,
+          lastName,
+          oldEmail,
+          newEmail,
+          undergrad,
+          grad,
+          employer,
+          position,
+          location,
+          linkedin,
+        },
+      }
+    ) {
+      const { errors, valid } = validateEditAlumniProfileInput(
+          firstName,
+          lastName,
+          oldEmail,
+          newEmail,
+          undergrad,
+          grad,
+          employer,
+          position,
+          location,
+          linkedin,
+      );
+
+      var coordinates = {
+        latitude: 0,
+        longitude: 0
+      };
+
+      if (!valid) {
+        handleInputError(errors);
+      }
+
+      grad.year = grad.year === "" ? 0 : grad.year;
+      var email = oldEmail;
+
+      if(newEmail !== oldEmail && newEmail !== ""){
+        const isEmailDuplicate = await Alumni.findOne({ email: newEmail });
```

Comment:
instead of using findOne here, use .countDocuments(/*options*/) [link](https://mongoosejs.com/docs/api/model.html#Model.countDocuments()).
This reduces the data that mongoose has to transmit (whether or not the doc exists vs the entire doc itself).

Reply:
Got rid of this query altogether. Since email is tagged as unique in the schema, findOneAndReplace() will return an error with code 11000 if the email is already in the DB.